pkg/build/controller/build/build_controller.go

```diff
@@ -1182,6 +1189,16 @@ func (bc *BuildController) createBuildPod(build *buildv1.Build) (*buildUpdate, e
 		}
 	}
 
+	// Check that ConfigMaps and secrets specified by build Source exist in the project
+	// if they don't exist, fail the build with GenericBuildFailed reason and
+	// corresponding message
+	if update, err := bc.checkConfigMapExists(build); err != nil {
+		return update, err
+	}
+	if update, err := bc.checkSecretExists(build); err != nil {
+		return update, err
+	}
```

Comment:
The check functions should return a list of missing resources, which then get converted to a BuildUpdate.